tests/test_benchmark.py

```diff
@@ -41,6 +42,18 @@ class BenchmarkConfig:
     num_batches: int
     steps_per_batch: int
     verbose: bool
+    generate_plots: bool
+
+
+def plot_batch_times(batch_times: List[float], label: str):
```

Comment:
uncertainty: more informative to plot batch times, or steps/second ?

Reply:
Probably most informative to plot ns per day as that is the final value reported and this will give us a sense of how much that number of interest jumps around.

Reply:
Plotting the ns per day for each batch timing: https://github.com/proteneer/timemachine/pull/1375/commits/b24d627a28b49a9d943d91e6d830e3817e7692c3

Reply:
Nice. Steps per second and maybe box volume could also be relevant.

(I assume the benchmark issue is still that the box volume is decreasing -- aka the number of particle pairs within cutoff is increasing -- relative to initialization. Maybe it's some other issue though.)

Reply:
The issue is just that I am benchmarking on a machine with other work going on. Added this code to be sure that when I do this 'for real' that the values are well converged.

Reply:
Don't think at the moment the steps per second or box volume are particularly informative. The steps per second is just the same value scaled differently and at least how I use this script is to maximize ns per day so seems best to plot that value. As for box volume it doesn't seem to matter based on what I have seen running this code on a machine without any GUI also based on work in https://github.com/proteneer/timemachine/pull/1368 the tile density doesn't change very much even after running the barostat (which surprised me)

Reply:
For some of the benchmark conditions (e.g. the ones where the box is initialized at an atypical volume), box volume appears highly correlated with speed

![image](https://github.com/user-attachments/assets/e10d0232-df32-4205-bae4-da012851a08d)

but for others (e.g. in conditions omitting a barostat), less so:

![image](https://github.com/user-attachments/assets/08b4d52f-9551-4854-ba88-6a6ec4056e18)

(Note: bad benchmarking practice -- I did not control for possible variation in clock rate over time here -- nvidia docs suggest to do so: https://developer.nvidia.com/blog/advanced-api-performance-setstablepowerstate/ )

-------------

Personally, I have found the inclusion of box volume to provide informative context on timings here, since one of the first things I might want to branch on is something like, "are the timings explainable by some factor that influences overall particle density". (Also, any big fluctuations or transients are useful to know about when assessing small measured differences in ns/day.)

If timing drifts like the one shown in the PR description persist after controlling for possible variation in clock speed (and ruling out variations in overall particle density), that could be suggestive of something else equilibrating slowly within the benchmark simulation boxes, in a way that the hilbert sort/etc. performance assumptions are sensitive to...

Reply:
Added plotting of box volumes in https://github.com/proteneer/timemachine/pull/1375/commits/64dfe6d53f3eaa06f83da45e3ca0640fbb8b16ae

Also added a CLI parameter to increase the number of batches run upfront to be able to do some burn in: https://github.com/proteneer/timemachine/pull/1375/commits/a14cda6386b78a26a562316f32af8f829d03dc3a